### DISCUSSION TITLE
Allow for changing osu!taiko note colors within skin.ini

### DISCUSSION THREAD

**Discussion by Azzilotl** [ppy/osu#29530] [Ideas]
Currently, as of right now, the colors of osu!taiko's notes are hardcoded with no real ways to truly customize them, making them stuck at the default red and blue.
![image](https://github.com/user-attachments/assets/3b0a02c8-ffa1-41ea-b4b9-16c59225979f)
Allowing the colors to be changed by the selected skin would bring more customizability to taiko skinning and could also help potential colorblindness or other issues that may harm the player in properly distinguishing the colors between the two notes.

This idea has also gotten a healthy amount of community support behind it, most being within a [9 year old post on the feature requests forums](https://osu.ppy.sh/community/forums/topics/328212?n=1), with a priority status of 1,173 (as of writing this).

To prevent issues with splitting (as both notes use the same texture) and skinning the note textures themselves, a new section for taiko could be added within skin.ini to change these colors similarly to how it would be done with the other rulesets. (rough example below)
`[Taiko]`
`ColourDon: 231, 237, 0`
`ColourKat: 150, 0, 224`
![image](https://github.com/user-attachments/assets/83022abe-a34b-4aca-9d09-899d091592d6)

**Comment by bdach** [comment]
One issue with this is that people could do what the "swap" mod does without using the mod. In stable this didn't matter, but now that that mod exists, it might.

The other thing I'm not sure about is whether this has any gameplay fairness implications. I don't think it should but I've been wrong on this before.

@Hiviexd do you have an opinion to offer on this?

**Reply by Hiviexd** [reply]
so far I can't imagine any downsides to this apart from the "swap" mod concern, which can be alleviated by treating it the same as osu!mania's "mirror" mod (ranked + 1x multiplier), that way it effectively doesn't matter if people use the "swap" mod or change the skin's colors.

**Comment by exformation** [comment]
Any update on this? This would make learning multiple playstyles a LOT easier. Like using red/blue for kddk and yellow/purple for ddkk.
